Pass release on to the sentry-server

fizker opened the pull request:
The `Event` class had support for a ReleaseVersion string, but there was no way to set it outside of calling `SentryClient.captureEvent(_:)`.

I have added it as an ivar to `SentryClient`, so that it will be sent automatically in case of crashes.

joshdholtz commented on Sources/Sentry.swift:
```diff
@@ -96,7 +99,7 @@ import Foundation
 	*/
 	@objc public func captureMessage(message: String, level: SentrySeverity = .Info) {
-		let event = Event(message, level: level)
+		let event = Event(message, level: level, release: releaseVersion)
```
@fizker Instead of here, the release should probably added into the `captureEvent` function inside of this if statement - https://github.com/revisohq/sentry-swift/blob/3c56b271ac9e40dcda84a1ce95c51e16a50ce7a0/Sources/Sentry.swift#L119-L129

joshdholtz commented:
@fizker Good call on adding this 👌 📞  One comment about moving something around.

Can you also add tests for this? 😉

joshdholtz commented:
@fizker My apologies! I just made a decent size merge and there are no conflicts 🙈  You will have to rebase 😇

fizker commented:
I rebased on master and updated according to your comment.

I can't add tests because I don't know how to install the dependency in a way that makes the project compile.

joshdholtz commented:
@fizker My apologies! I should put instructions in the readme on how to download the dependencies. You have to run `carthage update --no-build` or `git submodule update --init`

fizker commented:
@joshdholtz Great, that made the tests build and run :). I will look at writing some tests within the week, as I find the time

joshdholtz commented:
@fizker Glad you got it working! I will keep an eye out for your updates 😊

fizker commented:
I have added a few tests, but I want to align visions before going further ahead. The current tests go against some common best practices, which might make sense to align with going forward:

1. The tests were all put into a single file. I couldn't find any existing tests for the `SentryClient` class, so I added a new file explicitly for tests of this class.
2. `assert()` is used in preference of the `XCTAssert*()` family. `XCTAssert*()` gives better feedback and does not stop test execution at the first failure.
3. I prefer naming tests along the pattern of `actionToTest_prerequisites_expectedResult`. This is not as much a best practice as just my personal preference.

Please look it over and comment :)

joshdholtz commented on SentrySwiftTests/SentryClientTests.swift:
```diff
@@ -0,0 +1,65 @@
+import XCTest
+@testable import SentrySwift
+
+class TestCrashHandler:CrashHandler {
```
This is super cool 🙌

joshdholtz commented:
@fizker This looks solid 👍 I actually really did like that test naming. I will make a note to split up and improve the naming/testing for the other tests 😊

joshdholtz commented:
@fizker Also, thanks for the contribution ❤️

fizker commented:
Cool. I was totally not done writing tests, I just didn't want to go too far down that path if you disagreed on the style ;). I will probably add some tests regarding passing the `releaseVersion` on to the event object.

Re: the test naming, I use a snippet in Xcode that triggers on the word `test` and inserts the following code:
```swift
func test_<#func#>_<#context#>_<#expected#>() {
		<#body#>
	}
```
<img width="426" alt="test-snippet" src="https://cloud.githubusercontent.com/assets/736545/16107966/55886eca-33a0-11e6-8e57-5291a79ba92b.png">

This makes it much more convenient to follow the style.

joshdholtz commented:
@fizker Oops, my bad! I thought you were done 😇 I will gladly accept another PR with more tests though if you are still down for that 😉
